### PR TITLE
enclave-tls: Fix memory leak in enclave-tls-server

### PR DESCRIPTION
**Pull request by uudiin**
Each enclave_tls_negotiate() call will assign a value to the ssl
field in the tls handle. This value is generally the data allocated
by SSL_new(). After multiple calls, the previously allocated ssl
will be leaked because it is overwritten by the later value. Use
a new tls handle for each handshake to solve this issue.

### CODE REVIEW

**Review by jiazhang0 — APPROVED** [review]

**Review by haosanzi — APPROVED** [review]

File: enclave-tls/samples/enclave-tls-server/server.c [hunk 1]
```diff
@@ -184,13 +184,12 @@ int enclave_tls_server_startup(int sockfd, enclave_tls_log_level_t log_level, ch
...
 		close(connd);
+		enclave_tls_cleanup(handle);
 	}
 
 	return 0;
 
 err:
-	/* Ignore the error code of cleanup in order to return the prepositional error */
-	enclave_tls_cleanup(handle);
 	return -1;
```
Comment (YangLiang3) [inline enclave-tls/samples/enclave-tls-server/server.c:193]: why remove enclave_tls_cleanup when err happens?

Comment (uudiin) [reply enclave-tls/samples/enclave-tls-server/server.c:193]: `handle` has moved to while scope already.

Comment (YangLiang3) [reply enclave-tls/samples/enclave-tls-server/server.c:193]: I mean during session once something is wrong, the enclave_tls_clean will be invoked, goto err should use enclave_tls_clean to release some memory

Comment (uudiin) [reply enclave-tls/samples/enclave-tls-server/server.c:193]: This should be done in real scenarios, but if an error occurs in the sample, it means exiting. In order to simplify the implementation, let the process exit to clean up resources.

**Review by YangLiang3 — APPROVED** [review]

**Pull request merged**